frontend/src/component/changeRequest/ChangeRequest/Changes/Change/StrategyChange.tsx

```diff
@@ -119,7 +117,157 @@ const EditHeader: VFC<{
     return <Typography>Editing strategy:</Typography>;
 };
 
-export const StrategyChange: VFC<{
+const hasDiff = (object: unknown, objectToCompare: unknown) =>
+    JSON.stringify(object) !== JSON.stringify(objectToCompare);
+
+const DeleteStrategy: FC<{
+    change: IChangeRequestDeleteStrategy;
+    changeRequestState: ChangeRequestState;
+    currentStrategy: IFeatureStrategy | undefined;
+    actions?: ReactNode;
+}> = ({ change, changeRequestState, currentStrategy, actions }) => {
+    const name =
+        changeRequestState === 'Applied'
+            ? change.payload?.snapshot?.name
+            : currentStrategy?.name;
+    const title =
+        changeRequestState === 'Applied'
+            ? change.payload?.snapshot?.title
+            : currentStrategy?.title;
+    const referenceStrategy =
+        changeRequestState === 'Applied'
+            ? change.payload.snapshot
+            : currentStrategy;
+
+    return (
+        <>
+            <ChangeItemCreateEditDeleteWrapper className='delete-strategy-information-wrapper'>
+                <ChangeItemInfo>
+                    <Typography
+                        sx={(theme) => ({
+                            color: theme.palette.error.main,
+                        })}
+                    >
+                        - Deleting strategy:
+                    </Typography>
+                    <StrategyTooltipLink name={name || ''} title={title}>
+                        <StrategyDiff
+                            change={change}
+                            currentStrategy={referenceStrategy}
+                        />
+                    </StrategyTooltipLink>
+                </ChangeItemInfo>
+                <div>{actions}</div>
+            </ChangeItemCreateEditDeleteWrapper>
+            {referenceStrategy && (
+                <StrategyExecution strategy={referenceStrategy} />
+            )}
+            <ConditionallyRender
+                condition={Boolean(referenceStrategy?.variants?.length)}
+                show={
+                    referenceStrategy?.variants && (
+                        <StyledBox>
+                            <StyledTypography>
+                                Deleting strategy variants:
+                            </StyledTypography>
+                            <EnvironmentVariantsTable
+                                variants={referenceStrategy.variants}
+                            />
+                        </StyledBox>
+                    )
+                }
+            />
+        </>
+    );
+};
+
+const UpdateStrategy: FC<{
+    change: IChangeRequestUpdateStrategy;
+    changeRequestState: ChangeRequestState;
+    currentStrategy: IFeatureStrategy | undefined;
+    actions?: ReactNode;
+}> = ({ change, changeRequestState, currentStrategy, actions }) => {
+    const hasVariantDiff = hasDiff(
```

Comment:
extracted component can have display data calculation outside of the inline calls